### DISCUSSION TITLE
CDC source code browser? Where?

### DISCUSSION THREAD

**Discussion by KC5VDJ** [kej715/DtCyber#143] [General]
Been wondering this for a while...

Where is the hypertext browser that they obviously had?

If they didn't, internal cross-references in the internal documentation seem made for it.

**Comment by kej715** [comment]
Just in case this question wasn't asked tongue-in-cheek ...  CDC did not
have a hypertext source code browser.

**Comment by KC5VDJ** [comment]
Was wondering about the \*tags\*.

**Comment by kej715** [comment]
Tags? Please provide an example of what you mean by "tags"?

**Reply by KC5VDJ** [reply]
In the comments, for example near the beginning of 2.8.7's COMCMAC:

Under acquire:

"See \*QAC\* documentation for parameter block description."

I was thinking that CODING 2.2.2 was for the purpose, and it can be.

Maybe something gopher-like or lynx/links-like.  Starting with such tags, then working sideways on the "reserved" question using the same parser expanded.

I mentioned something about some sort of "assigned numbers" a while back, this would sort of shoehorn into that category.  (RIP, Jon, a really hoopy frood)

The "reserved" question would most likely need this for a definitive answer on which were truly RFU, and which are labeled as such, but used.  Any future distributed os enhancement by anyone would kinda need this rather than force people to give up their own uses for "installation reserved".  Parsing the CODING 2.2.2 "tags" does a bit of the grunt work anyway...it would make sense if I were to expand it for a deeper code analysis to track references.

How many million lines of code? I'd rather write a parser for this, as it will catch things missed by eye, even with inconsistencies with CODING.

**Reply by KC5VDJ** [reply]
The biggest problem would be consistency with CODING 2.2.2, as things stand in the source tapes.  I also see deviation within the same files even.  But it would be a start (after current project).

**Reply by KC5VDJ** [reply]
Could expand it to easily deal with a problem I have encountered with some (not many) card numbers being a bit to the left of where they belong...although that could be putty/winblowz dropping spaces, but...

**Comment by kej715** [comment]
The "*QAC*" in "See *QAC* documentation for parameter block description."
is not a hypertext tag. CDC coding style simply used * as a quotation mark.
You will not find the character " or the character ' used as a quotation
mark in any CDC-written OS code. You will always see * instead. That
statement in COMCMAC is simply emphasizing that QAC is the formal name of
an operating system component.

**Reply by KC5VDJ** [reply]
Came across CODING 2.2.2 last night, third paragraph.  See my revised comments above.

**Reply by KC5VDJ** [reply]
What I am saying is that it could be used as the basis of a hyperlinked source base.

Then added that the same code could be expanded for other purposes in addition to.

**Comment by kej715** [comment]
What is "CODING 2.2.2"? Google searches for this term don't seem to return
any results related to hypertext or documentation systems. Also, have you
seen William Schaub's work on hypertext-ized NOS source code?

**Reply by KC5VDJ** [reply]
In 2.8.7

attach,coding/un=library.

There is also a printed version in the bitsavers collection.  With an internal memo as a preface, iirc.

**Reply by KC5VDJ** [reply]
Actually, I wasn't aware of his efforts.

**Reply by KC5VDJ** [reply]
And Google, with rare exception, is practically worthless when it comes to Control Data.  Too much name clash, and dumbed down searching (someone offering web search using actual Boolean searches....cha-ching!)
